src/lib/ProjectM36/Client/Simple.hs

```diff
@@ -0,0 +1,98 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+module ProjectM36.Client.Simple (
+  simpleConnectProjectM36,
+  withTransaction,
+  execute,
+  query,
+  rollback,
+  close,
+  Atom(..),
+  AtomType(..),
+  DBError(..),
+  Attribute(..),
+  C.ConnectionInfo(..),
+  C.PersistenceStrategy(..),
+  C.NotificationCallback,
+  C.emptyNotificationCallback,
+  C.DatabaseContextExpr(..),
+  C.RelationalExprBase(..)  
+  ) where
+-- | A simplified client interface for Project:M36 database access.
+import ProjectM36.Error
+import ProjectM36.Base
+import qualified ProjectM36.Client as C
+import Control.Monad.Reader
+import Control.Exception.Base
+
+type DBConn = (C.SessionId, C.Connection)
+
+newtype DB a = DB {runDB :: ReaderT (C.SessionId, C.Connection) IO a}
```

Comment:
If we later decide to add a monad transformer, this will become `DBT` which seems pretty hard to read. Perhaps using `Db` (lower case `b`) would be better prep for this (`DbT` isn't as bad). I personally prefer not to use all-caps for acronyms anyway, for this reason. But it's your call.

Reply:
Why not use `DBConn` here as well?

Reply:
Also we discussed at one point tracking read-only vs write-access in this monad. E.g. you could have a phantom type: `newtype Db mode = Db ...`. And then `data ReadOnly` and `data ReadWrite`. `query` would be polymorphic over this type and `execute` would be monomorphic to `ReadWrite`. This would prevent `execute` from ever occurring in a `ReadOnly` transaction.

Another advantage is that people could write their own access modes and use type families and such to guard access to things in the database.

This is kinda cool...but do we need it?

Reply:
It's worth noting that because you have the lower level API as well, anyone can write their own high-level API on top. I could provide this feature as a library instead.

Reply:
If we have read-only transactions, I would like that to be enforced on the server-side as well. Can we add it later?

Reply:
Yeah. If I want it I can always add it to my app code very easily.